src/auth/basic/Config.cc

```diff
@@ -176,6 +295,18 @@ Auth::Basic::Config::decodeCleartext(const char *httpAuthHeader)
     if (base64_decode_update(&ctx, &dstLen, reinterpret_cast<uint8_t*>(cleartext), srcLen, eek) && base64_decode_final(&ctx)) {
         cleartext[dstLen] = '\0';
 
+        if (!isLegalUTF8String(cleartext, cleartext + dstLen)) {
+            SBuf str;
+
+            if (isCP1251EncodingAllowed(request))
```

Comment:
This is converting the nominally 'is allowed' test function into a 'force to happen' test because the else clause Latn-1 conversion is skipped whenever cp1251 is merely _allowed_.

As Alex mentioned relying on languages is somewhat problematic. While this may work for you it is not guaranteed to be like that everywhere. At least for sr-* we have sr-Latn alphabet translations for our error pages because that alphabet is actively being used by large populations (plural, some outside the country itself).

The same might be true for any of the languages, just we do not have evidence supporting.  That means when the cp-1251 does not succeed we still may need to try the Latin-1 conversion.

Reply:
FTR the Cp1251ToUtf8 was updated to not have error cases other than NULL input. The output may be garbage but there is no longer a need for and should not be any followup conversion now.